judge/views/problem.py

```diff
@@ -213,8 +226,117 @@ def get_context_data(self, **kwargs):
                                           context['description'], 'problem')
         context['meta_description'] = self.object.summary or metadata[0]
         context['og_image'] = self.object.og_image or metadata[1]
+
+        context['can_vote'] = self.can_vote(user, self.object)  # if this problem is votable by this user
+        # the vote this user has already cast on this problem
+        if context['can_vote']:
+            vote = ProblemPointsVote.objects.filter(voter=user.profile, problem=self.object)
+        # whether or not they've already voted
+        context['has_voted'] = context['can_vote'] and vote.exists()
+        if context['has_voted']:
+            context['voted_points'] = vote.first().points  # the previous vote's points
+            context['voted_note'] = vote.first().note
+
+        if 'problem_points_vote_form' not in context:
+            context['problem_points_vote_form'] = ProblemPointsVoteForm({})
+
+        if 'has_errors' not in context:
+            context['has_errors'] = False
+
+        if 'points_placeholder' not in context:  # placeholder for the points
+            if context['has_voted']:  # if voted, the vote
+                context['points_placeholder'] = context['voted_points']
+            else:  # otherwise a *nice* default
+                context['points_placeholder'] = 69.42069
+
+        if 'note_placeholder' not in context:
+            if context['has_voted']:
+                context['note_placeholder'] = context['voted_note']
+            else:
+                context['note_placeholder'] = self.default_note()
+
+        all_votes = sorted([v.points for v in ProblemPointsVote.objects.filter(problem=self.object)])
+        context['has_votes'] = len(all_votes) > 0
+        if context['has_votes']:
+            context['mean_vote'] = sum(all_votes) / len(all_votes)
+            context['num_votes'] = len(all_votes)
+            context['min_vote'] = all_votes[0]
+            context['max_vote'] = all_votes[-1]
+
+            # provides index and value of the median of some range of the data
+            def median(left_index, right_index, data):
+                size = right_index - left_index + 1
+                index = left_index + (size - 1) / 2
+                value = None
+                if size % 2 == 1:
+                    value = data[int(index)]
+                else:
+                    value = (data[math.floor(index)] + data[math.ceil(index)]) / 2
+                return index, value
+
+            median_data = median(0, len(all_votes) - 1, all_votes)
+            context['median_vote'] = median_data[1]
+            median_index = median_data[0]
+
+            context['enough_data_for_plot'] = len(all_votes) > 2
```

Comment:
Can we find a way to draw a plot even if there are less than 2 votes? Perhaps the median function could be changed to a function that accepts a quartile number (1, 2, 3), and computes the index based on that. Tiebreaking might not be necessary (just floor may be enough for plot purposes?), but if you really want, perhaps you could do some weighted average between the floor and ceil indices?

Reply:
The median and mean are displayed regardless of whether or not there is enough data to form a box and whisker plot. In the case of one data point it is unnecessary to display a box plot for this reason. 
However, if there are two data points, I can modify it to return the quartiles as equivalent to their adjacent extrema, which should work with the template.